### PR TITLE
Migrate tasks to use `childManagerAccountID`

### PR DESCRIPTION
**Pull request by Beamanator**
~currently holding on https://github.com/Expensify/Web-Expensify/pull/37963~

### Details
Just starting to use the new accountID property instead of email, in order to limit the emails we're passing around the app. This has been available for a few weeks but forgot to migrate this.

### Fixed Issues
$ Part of https://github.com/Expensify/App/issues/19007 (wasn't noticed before)

### Tests
Tests related to https://github.com/Expensify/App/pull/19990:

#### Tests for https://github.com/Expensify/App/issues/19089

1. Login with Expensifail account
2. Go to FAB > Assign task
3. Enter title, description and click on next button
4. Click on assignee  and select any user or enter an email or phone number to select a new user if don't have any user
5. Click on confirm button to create a new task
6. Verify that the assigned user displays as mention in the task preview
7. Edit the title of this task
8. Logout and login again
9. Verify that the assigned users still displays as mention in the task preview as well

#### Tests for https://github.com/Expensify/App/issues/19959
1. Go to App and Click on + icon
2. Select Assign task option
3. Create a new task
4. Go to settings > logout
5. Login again
6. Verify that the task name is created in above display

#### Tests for https://github.com/Expensify/App/issues/19967
1. Go to any chat
2. Click on + icon and select Assign task option
3. Enter title and description
4. Click on Next button
5. Click on confirm task button
6. click on title
8. Edit task title
10. Go back to previous report screen
11. Verify that Task title is updated

Make sure this still works (from [this recent PR](https://github.com/Expensify/App/pull/21659)):
1. Go to any chat (1:1 & group at least)
2. Click + button in composer
3. Click Assign task
4. Input a title => Click next
5. Verify that by default the "share somewhere" is in that chat, and will not set an assignee by default and no error message
6. Click Confirm task.
7. Verify that the task is created with no assignee & it doesn't show an empty avatar

- [ ] Verify that no errors appear in the JS console

### Offline tests
Same as above

### QA Steps
Same as above

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

Screenshot 2023-06-29 at 3 14 23 PM

</details>

<details>
<summary>Mobile Web - Chrome</summary>

</details>

<details>
<summary>Mobile Web - Safari</summary>

</details>

<details>
<summary>Desktop</summary>

Screenshot 2023-06-29 at 3 17 50 PM

</details>

<details>
<summary>iOS</summary>

</details>

<details>
<summary>Android</summary>

Screenshot 2023-06-29 at 3 24 32 PM

</details>

### CODE REVIEW

File: src/pages/workspace/WorkspaceMembersPage.js
Comment (Beamanator) [inline src/pages/workspace/WorkspaceMembersPage.js:1]: Noticed this needed to be updated, maybe it belongs in a different PR? 🤷

**Comment by Beamanator** [comment]
Updated test steps & added screenshots - I thinkkkkk we could get a C+ to help test this, once the web-e PR is on staging :D

**Comment by Beamanator** [comment]
feel free to get a C+ to help review if you'd like!

also cc @puneetlath in case you're free to review

**Review by bondydaa — CHANGES_REQUESTED** [review]
does this instance need to be removed as well? https://github.com/Expensify/App/blob/8b7a65dcd36fa2b47ffb39c69f2cb294b94a7171/src/libs/ReportUtils.js#L1263

**Comment by Beamanator** [comment]
Thanks @bondydaa , good find 🤦

**Review by bondydaa — DISMISSED** [review]

**Comment by bondydaa** [comment]
asked in contributor plus to get someone to test and do the checklist for us.

**Review by 0xmiroslav — CHANGES_REQUESTED** [review]
Bug (regression): task assignee not displaying in TaskPreview

Repro step:
Assign task to any random user (not in contact list) in any DM chat

Before:

https://github.com/Expensify/App/assets/97473779/a09eb71e-b4ed-49e0-bded-7f4807961506


After:

https://github.com/Expensify/App/assets/97473779/a7f1a3dc-d9aa-41da-9ce8-0245cfe09179

File: src/components/ReportActionItem/TaskPreview.js [hunk 1]
```diff
@@ -59,7 +65,8 @@ function TaskPreview(props) {
         ? props.taskReport.stateNum === CONST.REPORT.STATE_NUM.SUBMITTED && props.taskReport.statusNum === CONST.REPORT.STATUS.APPROVED
         : props.action.childStateNum === CONST.REPORT.STATE_NUM.SUBMITTED && props.action.childStatusNum === CONST.REPORT.STATUS.APPROVED;
     const taskTitle = props.taskReport.reportName || props.action.childReportName;
-    const taskAssignee = props.taskReport.managerEmail || props.action.childManagerEmail;
+    const taskAssigneeAccountID = TaskUtils.getTaskAssigneeAccountID(props.taskReport);
```
[the remark this answers is not in the document]
Comment (Beamanator) [reply src/components/ReportActionItem/TaskPreview.js:68]: I don't think so - but since we were defaulting to `managerEmail` first, I thought it would be best if we default to `managerID` first as well, THEN use `childManagerAccountID` - and that logic exists in this function so I thought it would be better to reuse it, do you disagree?

File: src/pages/workspace/WorkspaceMembersPage.js
Comment (Beamanator) [reply src/pages/workspace/WorkspaceMembersPage.js:1]: You can see I'm just including extra prop types / default prop types that were missing. We used `withCurrentUserPersonalDetails` already, but didn't declare the prop types.

**Comment by Beamanator** [comment]
Checking that regression you mentioned @0xmiroslav 👍

File: src/components/ReportActionItem/TaskPreview.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Beamanator) [reply src/components/ReportActionItem/TaskPreview.js:68]: "original logic" being `props.taskReport.managerEmail || props.action.childManagerEmail;`? We're trying to get rid of those :D

**Comment by Beamanator** [comment]
Ya so it kinda makes sense that the mentioned user doesn't exist the way we have it right now because I temporarily forgot `login` won't exist for these brand new users, since you don't "know" them.

So the possible changes are:
1. We try to show `login`, then as a backup show `displayName`
2. We go directly to showing `displayName` in the mention

cc @puneetlath @thienlnam any preferences here? Tagging you both since this is a combo of Task & Mention feature

**Comment by puneetlath** [comment]
Hm yeah that's interesting. We're going to migrate mentions to be accountID-based but it's just not quite done yet. So whatever we do for now will be temporary. I think we can just do whatever is easiest.

**Comment by Beamanator** [comment]
Awesome, then I will probably just switch to defaulting by displayName for now 👍

**Comment by Beamanator** [comment]
@0xmiroslav ready for re-review please 🙏

**Comment by Beamanator** [comment]
@0xmiroslav if neither exist then `htmlForTaskPreview` will just be `<comment>${taskTitle}</comment>`, no?

**Review by 0xmiroslav — APPROVED** [review]
LGTM!

**Review by puneetlath — APPROVED** [review]

File: src/components/ReportActionItem/TaskPreview.js [hunk 2]
```diff
@@ -65,7 +66,7 @@ function TaskPreview(props) {
         : props.action.childStateNum === CONST.REPORT.STATE_NUM.SUBMITTED && props.action.childStatusNum === CONST.REPORT.STATUS.APPROVED;
     const taskTitle = props.taskReport.reportName || props.action.childReportName;
     const taskAssigneeAccountID = TaskUtils.getTaskAssigneeAccountID(props.taskReport);
-    const taskAssignee = lodashGet(props.personalDetailsList, [taskAssigneeAccountID, 'login'], '');
+    const taskAssignee = lodashGet(props.personalDetailsList, [taskAssigneeAccountID, 'login'], lodashGet(props.personalDetailsList, [taskAssigneeAccountID, 'displayName'], ''));
```
Comment (bondydaa) [inline src/components/ReportActionItem/TaskPreview.js:69]: personally not crazy about this nested lodash, would rather see something like:
```suggestion
    const assigneeLogin = lodashGet(props.personalDetailsList, [taskAssigneeAccountID, 'login'], '');
    const assigneeDisplayName = lodashGet(props.personalDetailsList, [taskAssigneeAccountID, 'displayName'], '');
    const taskAssignee = assigneeLogin || assigneeDisplayName;
```

as it's more clear to me that `displayName` is a fall back if we don't have a login. instead of trying to reason how the 2nd lodashGet might get called and be a fallback for the first call.

**Review by bondydaa — APPROVED** [review]
NAB from me but if you want to clean it up then bonus points 😂

File: src/components/ReportActionItem/TaskPreview.js (hunk 2 above)
Comment (Beamanator) [reply src/components/ReportActionItem/TaskPreview.js:69]: Ooh that's totally fair - I'll try to remember to clean this up later - i can see it being useful but not highest priority at the moment

Comment (puneetlath) [reply src/components/ReportActionItem/TaskPreview.js:69]: Cleanup PR here: https://github.com/Expensify/App/pull/22439

File: src/components/ReportActionItem/TaskPreview.js (hunk 1 above)
Comment (mananjadhav) [inline src/components/ReportActionItem/TaskPreview.js:68]: We didn't consider the case where the `taskReport` could be empty and hence it caused a regression [here](https://github.com/Expensify/App/issues/22373), because the task assignment didn't show an email address.

Comment (Beamanator) [reply src/components/ReportActionItem/TaskPreview.js:68]: hmm if task report could be empty, would this have crashed before too? 🤔 (since we used to directly access `props.taskReport.managerEmail`
